lightning/src/ln/peer_channel_encryptor.rs

```diff
@@ -403,6 +418,42 @@ impl PeerChannelEncryptor {
 		res
 	}
 
+	/// Encrypts the given message, returning the encrypted version.
+	/// panics if the length of `message`, once encoded, is greater than 65535 or if the Noise
+	/// handshake has not finished.
+	pub fn encrypt_message<M: wire::Type>(&mut self, message: &M) -> Vec<u8> {
```

Comment:
Could we consolidate this with `encrypt_buffer` if we made `gossip_broadcast_buffer` contain messages instead of encoded messages?